### PR TITLE
Added possibility to use second ENTSOE token

### PR DESCRIPTION
**Pull request by robertahunt**
Is there a better way to do this?

### CODE REVIEW

File: parsers/ENTSOE.py [hunk 1]
```diff
@@ -407,7 +407,13 @@ def query_ENTSOE(session, params, target_datetime=None, span=(-48, 24)):
     params['periodEnd'] = target_datetime.shift(hours=span[1]).format('YYYYMMDDHH00')
     if 'ENTSOE_TOKEN' not in os.environ:
         raise Exception('No ENTSOE_TOKEN found! Please add it into secrets.env!')
-    params['securityToken'] = os.environ['ENTSOE_TOKEN']
+    tokens = [os.environ['ENTSOE_TOKEN']]
```
Comment (corradio) [inline parsers/ENTSOE.py]: Suggestion:
```suggestion
    tokens = os.environ['ENTSOE_TOKEN'].split(',')
```

That way we can pass one or multiple ENTSOE_TOKEN separated by a comma

**Comment by corradio** [comment]
Do we also need to change the parser in entsoe-tp-app or does it use contrib?

**Comment by robertahunt** [comment]
> Do we also need to change the parser in entsoe-tp-app or does it use contrib?

I don't know... I don't even know how I would find out.

**Comment by fbarl** [comment]
Interesting - so the cause of #2257 was crossing the query limits to the ENTSOE API?

Their [API docs](https://transparency.entsoe.eu/content/static_content/Static%20content/web%20api/Guide.html#_query_limits) say:

> Max 400 requests are accepted per user per minute.

I'm not very familiar with how the parsers are currently being employed, but if that's true, that should be enough to stay within limits if their responses were cached on the backend - is there any sort of caching mechanism in place right now? :thinking:

**Comment by corradio** [comment]
Sent a private email

**Comment by AyrtonB** [comment]
Have you tried just adding in something like `sleep(0.5)` in between calls?

**Comment by robertahunt** [comment]
@AyrtonB I think we run all the parsers in parallel so I don't think this will solve it on our production server... that is what I am doing as part of my hacky fix though :P

**Comment by AyrtonB** [comment]
How are the parsers parallelised? If they're always split into batches of the same domains you could have a dictionary mapping each of those batches to different keys, that could remove the chance of multiple parser operations randomly selecting the same key.

**Comment by robertahunt** [comment]
Haha I like the way you think! But i'd like to avoid adding too much complexity to our system unless we have to. First we are following up with ENTSOE to see if we can arrange something.

**Pull request merged**